plugin/pkg/admission/namespace/lifecycle/admission.go

```diff
@@ -45,11 +46,30 @@ func init() {
 type lifecycle struct {
 	*admission.Handler
 	client             clientset.Interface
-	store              cache.Store
 	immortalNamespaces sets.String
+	namespaceInformer  framework.SharedIndexInformer
+	// forceLiveLookupCache holds a list of entries for namespaces that we have a strong reason to believe are stale in our local cache.
+	// if a namespace is in this cache, then we will ignore our local state and always fetch latest from api server.
+	forceLiveLookupCache *lru.Cache
```

Comment:
https://github.com/kubernetes/kubernetes/pull/22951

the history of the original behavior to modify the underlying store was done in the earlier PR to address flakes with single API server.  It's understood that absent this change, in HA environments, there is still a race potential where API server 1 rejects a request (updated cache), and API server 2 accepts a request (stale cache), and the current background controller will hopefully observe and delete the late added resources as part of namespace clean-up. 

so given the original scope of the intended fix, I think a limited size LRU is hopefully sufficient.